planner/core/logical_plan_builder.go

```diff
@@ -699,7 +700,8 @@ func (b *PlanBuilder) coalesceCommonColumns(p *LogicalJoin, leftPlan, rightPlan
 }
 
 func (b *PlanBuilder) buildSelection(ctx context.Context, p LogicalPlan, where ast.ExprNode, AggMapper map[*ast.AggregateFuncExpr]int) (LogicalPlan, error) {
-	b.optFlag = b.optFlag | flagPredicatePushDown
+	b.optFlag |= flagPredicatePushDown
+	b.optFlag |= flagGcSubstitute
```

Comment:
Can we only  set this flag when build DataSource?

Reply:
Done